Content.Server/Fluids/EntitySystems/PuddleSystem.Transfers.cs

```diff
@@ -14,6 +15,12 @@ private void InitializeTransfers()
 
     private void OnRefillableDragged(Entity<RefillableSolutionComponent> entity, ref DragDropDraggedEvent args)
     {
+        if (!HasComp<HandsComponent>(args.User))
+        {
+            _popups.PopupEntity(Loc.GetString("mopping-system-no-hands"), args.User, args.User);
+            return;
+        }
+
```

Comment:
Not sure if you're still looking at this pr but either use .CanInteract (still lets stuff like dogs do it) or .CanComplexInteract (mostly based around hand stuff like computers).

May also need adding to the event itself to cache the value.